packages/amplify-graphql-conversation-transformer/src/transformer-steps/conversation-resolver-generator.ts

```diff
@@ -253,6 +267,31 @@ export class ConversationResolverGenerator {
     );
   }
 
+  /**
+   * Sets up the conversation table index
+   * @param ctx - The transformer context provider
+   * @param directive - The conversation directive configuration
+   */
+  private setUpConversationTableIndex(ctx: TransformerContextProvider, directive: ConversationDirectiveConfiguration): void {
+    const conversationName = directive.conversation.model.name.value;
+    const conversation = directive.conversation.model;
+
+    const conversationTable = getTable(ctx, conversation);
+    const gsiPartitionKeyName = '__typename';
+    const gsiPartitionKeyType = 'S';
+    const gsiSortKeyName = 'updatedAt';
+    const gsiSortKeyType = 'S';
+
+    this.addGlobalSecondaryIndex(
```

Comment:
I would have expected PK=ownerId; SK=updatedAt. What drives the decision to use type name rather than owner?

Reply:
That's also what I was hoping to do here -- it should* work great for queries originating from users (CUP / OIDC access token) because the resolver function knows the `owner` field value before executing the query. The prerequisite of knowing the `owner` field value would make the GSI unusable for admin (IAM) queries though. We could work around this with both GSIs and only using the owner+updatedAt index if identity context provides that info. I would like to do this, but I'm also hesitant to further bifurcate the user owner and admin query flows.

*should being the key word. I've often wondered why we don't automatically create a GSI where PK=ownerId on tables where the model is `owner` auth based. If there's a good reason, it's likely relevant here.